test/gpu/native/studies/transpose/transpose.chpl

```diff
@@ -0,0 +1,98 @@
+use CTypes;
+use GPUDiagnostics;
+use GPU;
+use Time;
+
+config const numTrials = 10;
+config const useNaive = false;
+config const sizeX = 2048*8,
+             sizeY = 2048*8;
+config param blockSize = 16;
+
+pragma "codegen for GPU"
+pragma "always resolve function"
+export proc transposeMatrix(odata: c_ptr(real(32)), idata: c_ptr(real(32)), width: int, height: int) {
+  // Allocate extra columns for the shared 2D array to avoid bank conflicts.
+  var smVoidPtr = __primitive("gpu allocShared", numBytes(real(32))*(blockSize+1)*blockSize);
+  var smArrPtr = smVoidPtr : c_ptr(real(32));
+
+  const blockIdxX = __primitive("gpu blockIdx x"),
+        threadIdxX = __primitive("gpu threadIdx x"),
+        blockIdxY = __primitive("gpu blockIdx y"),
+        threadIdxY = __primitive("gpu threadIdx y");
+  var idxX = blockIdxX * blockSize + threadIdxX,
+      idxY = blockIdxY * blockSize + threadIdxY;
+
+  // Store the input data in transposed order into temporary array
+  // i.e., the below is effectively smArrPtr[y][x] instead of
+  // smArrPtr[x][y] for copy
+  if (idxX < width && idxY < height) {
+    smArrPtr[(blockSize+1) * threadIdxX + threadIdxY] = idata[idxY * width + idxX];
+  }
+
+  // synchronize the threads
+  __primitive("gpu syncThreads");
+
+  // Swap coordinates and write back out
+  idxX = blockIdxY * blockSize + threadIdxX;
+  idxY = blockIdxX * blockSize + threadIdxY;
+  if (idxX < height && idxY < width) {
+    odata[idxY * height + idxX] = smArrPtr[(blockSize+1) * threadIdxY + threadIdxX];
+  }
+}
+
+proc transposeTiled(original, output) {
+  __primitive("gpu kernel launch",
+          c"transposeMatrix",
+          /* grid size */  sizeX / blockSize, sizeY / blockSize, 1,
+          /* block size */ blockSize, blockSize, 1,
+          /* kernel args */ c_ptrTo(output), c_ptrTo(original), sizeX, sizeY);
+}
+
+proc transposeNaive(original, output) {
+  foreach (x,y) in original.domain do output[y,x] = original[x,y];
+}
+
+on here.gpus[0] {
+  var original: [0..#sizeX, 0..#sizeY] real(32);
+  var output: [0..#sizeY, 0..#sizeY] real(32);
```

Comment:
One of the ways that I want to play with this benchmark is by changing the data type. I imagine, I'd probably switch between single and double precision reals, but maybe ints, too?

In any case, could you put a `config type dataType = real(32);` and use `dataType` here (and in bunch of other places) to make the test more ergonomic for study?